pytest: *BROKEN* in `test_segwit_shutdown_script`

### How to replicate 

On CLN v26.04rc3, using MacOS M5 Tahoe 26.4, run `uv run pytest -vvvv tests/test_closing.py::test_segwit_shutdown_script`

The test fail with the following error:

```
FAILED tests/test_closing.py::test_segwit_shutdown_script - pyln.client.lightning.RpcError: RPC call failed: method: connect, payload: {'id': '022c9f89ec272db20eb8ac5969f984d51c696bbf7b6bfd633c55801f7957c4d4be', 'host': 'localhost', 'port': 56821}, error: {'code': 401, 'message': 'All addresses failed: 127.0.0.1:56821: Cryptographic handshake: peer closed connection (wrong key?). '}
ERROR tests/test_closing.py::test_segwit_shutdown_script - ValueError: 
Node errors:
 - lightningd-31: had BROKEN or That's weird messages
 - lightningd-31: Node exited with return code 1
Global errors:
```



Broken log:

```
lightningd-31 2026-04-15T07:52:26.523Z **BROKEN** connectd: STATUS_FAIL_HSM_IO: No hsmd ECDH response
```
### Additional Context
Please note that the failing node is completely random. The `BROKEN` log shown above happened on `lightningd-31`, but in other runs, it fails on different nodes (e.g., 7, 8, 10, 14, 17, etc.).
On Linux the same test is passed.

--- a/common/ecdh_hsmd.c
+++ b/common/ecdh_hsmd.c
@@ -1,5 +1,6 @@
 #include "config.h"
 #include <assert.h>
+#include <ccan/io/io.h>
 #include <common/ecdh.h>
 #include <common/ecdh_hsmd.h>
 #include <common/utils.h>
@@ -33,4 +34,6 @@ void ecdh_hsmd_setup(int hsm_fd,
 {
 	stashed_hsm_fd = hsm_fd;
 	stashed_failed = failed;
+	/* Like read_fds in subd.c: don't trust sender's O_NONBLOCK state (issue #9060). */
+	io_fd_block(hsm_fd, true);
 }

